R/test-threads.R

```diff
@@ -151,46 +151,81 @@ check_run_times <- function(
     check_model_object(.mods, .mod_types = c("bbi_summary_list"))
   }
 
+  if(.wait){
+    tryCatch({
+      wait_for_nonmem(.mods, ...)
+    }, error = function(cond){
+      return(invisible(TRUE))
+    }, warning = function(cond){
+      message(cond)
+      message("\nConsider setting/increasing the `time_limit` in `wait_for_nonmem()`. See ?check_run_times for details")
+      return(invisible(TRUE))
+    })
+  }
 
-
-  tryCatch({
-    if(.wait) wait_for_nonmem(.mods, ...)
-
-    map_dfr(.mods, ~ {
+  map_dfr(.mods, ~ {
+    tryCatch({
```

Comment:
This works, but at this point I think we've built this section up so much that it's gotten a bit unwieldy. I think it can be refactored to simplify quite a bit, which will make maintaining it significantly easier.

Try replacing all of lines 167-213 with this:

```r
    tryCatch({
      # unpack bbi_summary_list element
      if (!is.null(.x$bbi_summary)) .x <- .x$bbi_summary

      # get model id and convert to summary object if necessary
      model_run <- get_model_id(.x)
      if (inherits(.x, NM_MOD_CLASS)) {
        .x <- model_summary(.x, .bbi_args = list(no_grd_file = TRUE, no_ext_file = TRUE, no_shk_file = TRUE))
      }

      # get number of threads from bbi_config.json
      config <- jsonlite::fromJSON(file.path(.x[[ABS_MOD_PATH]], "bbi_config.json"))
      threads <- ifelse(
        config$configuration$parallel,
        config$configuration$threads,
        1
      )

      data <- tibble::tibble(
        run = model_run,
        threads = threads,
        estimation_time = .x$run_details$estimation_time,
        covariance_time = .x$run_details$covariance_time,
        cpu_time = .x$run_details$cpu_time) %>%
        select(run, threads, all_of(.return_times))
      return(data)
```

Reply:
Tested that it worked with this change pretty diligently  and was pleasantly surprised. Cant believe you reduced it by that much, but yeah I agree it looks much nicer and will be much easier to evolve with time.